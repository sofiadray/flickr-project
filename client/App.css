body {
  text-align: center;
}

.App {
	text-align: center;
	padding: 2em;
}

.search {
  text-align: left;
  margin: .7em 0 .7em 0;
  padding: .7em;
  outline: none;
}

input {
	font-size: 15px;
	height: 1.5em;
	
}

input:active, input:focus {
	border-color: #1F2022;
}

.arrow {
	float:left;
	display: inline-flex;
	box-sizing: border-box;
	margin-left: auto;
	margin-right: auto;
	margin-top:15%;
}

i {
  padding: auto 3em auto 3em;
  color: #808183 ;
  transition-timing-function: ease-in-out;
  transition: .4s;
}

i:hover {
	color: #1F2022;
}

.jumbotron-container {
	margin-top: 2em;
	margin-bottom: 2em;
	width: 100%;
	text-align: center;
	display: flex;
}

.jumbotron-main {
	display: inline-block;
	margin-left: auto;
	margin-right: auto;
	text-align: left;
}


.thumbnail {
	opacity: .8;
	display: inline-block;
	margin-left: auto;
	margin-right: auto;
	text-align: left;
	transition-timing-function: ease-in-out;
	transition: .4s;
}

.thumbnail:hover {
	opacity: 1;
}

.thumbnail-row {
	display: block;
	text-align: center;
	width:100%;
	float: left;
	background-color: #E7E8EA;
	height:auto;
}

.jumbotron-main img {
-webkit-animation: fadein 2s; /* Safari, Chrome and Opera > 12.1 */
   -moz-animation: fadein 2s; /* Firefox < 16 */
    -ms-animation: fadein 2s; /* Internet Explorer */
     -o-animation: fadein 2s; /* Opera < 12.1 */
        animation: fadein 2s;
}

img {
	max-width: 100%;
    height: auto;
}

@keyframes fadein {
    from { opacity: 0; }
    to   { opacity: 1; }
}

/* Firefox < 16 */
@-moz-keyframes fadein {
    from { opacity: 0; }
    to   { opacity: 1; }
}

/* Safari, Chrome and Opera > 12.1 */
@-webkit-keyframes fadein {
    from { opacity: 0; }
    to   { opacity: 1; }
}

/* Internet Explorer */
@-ms-keyframes fadein {
    from { opacity: 0; }
    to   { opacity: 1; }
}

/* Opera < 12.1 */
@-o-keyframes fadein {
    from { opacity: 0; }
    to   { opacity: 1; }
}
