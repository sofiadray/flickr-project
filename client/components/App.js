import React, { Component } from 'react';
import PropTypes from 'prop-types';
import '../App.css';
import { getPhotos } from "../flickr.js";

var initialSlides = [
  'https://images.unsplash.com/photo-1444703686981-a3abbc4d4fe3?fit=crop&fm=jpg&h=825&q=80&w=1325',
  'https://images.unsplash.com/photo-1445251836269-d158eaa028a6?fit=crop&fm=jpg&h=825&q=80&w=1325',
  'https://images.unsplash.com/photo-1443926818681-717d074a57af?fit=crop&fm=jpg&h=825&q=80&w=1325',
  'https://images.unsplash.com/photo-1445251836269-d158eaa028a6?fit=crop&fm=jpg&h=825&q=80&w=1325',
];

class App extends Component {
    state = { 
      selected_image: initialSlides[0],
      slides: initialSlides,
      current_idx: 0
    }

  handleSearch = (event) => {
    let searchText = event.target.value
    if (searchText.length > 0) {
      getPhotos(searchText).then((data) => {
        this.setState({
          selected_image: data[0],
          slides: data,
          current_idx: 0
        })
      })
    }
  }

  displayThumbnail = (idx) => {
    this.setState({
      selected_image: this.state.slides[idx],
      current_idx: idx
    });  
  }

  showNextThumbnail = () => {
    if (this.state.current_idx + 1 === this.state.slides.length) {
      this.setState({
        current_idx:0,
        selected_image: this.state.slides[0]
      }); 
    }
    else {
      let index = this.state.current_idx + 1
      this.setState({ 
        current_idx: index, 
        selected_image: this.state.slides[index] 
      });
    }
  }

  showPrevThumbnail = () => {
    if (this.state.current_idx === 0 ) {
      let last = this.state.slides.length - 1
      this.setState({ 
        current_idx: last, 
        selected_image: this.state.slides[last]
      });
    }
    else {
      let index = this.state.current_idx - 1
      this.setState({ 
        current_idx: index, 
        selected_image: this.state.slides[index] 
      });
    }
  }

  render() {
    return (
      <div className="App">
          <SearchBar 
            handleSearch={this.handleSearch}
          />
          <Jumbotron 
            url={this.state.selected_image} 
            showNextThumbnail={this.showNextThumbnail} 
            showPrevThumbnail={this.showPrevThumbnail}
          />
          <ThumbnailRow 
            thumbnails={this.state.slides} 
            displayThumbnail={this.displayThumbnail} 
          />
      </div>
    );
  }
}

class SearchBar extends Component {
  static propTypes = {
    handleSearch: PropTypes.func.isRequired, 
  }
  render() {
    const { handleSearch } = this.props
    return (
      <input 
        type="text" 
        placeholder="Start Typing..." 
        className="search" 
        onChange={handleSearch}
      />
    );
  }
}

class Jumbotron extends Component {
  static propTypes = {
    url: PropTypes.string,
    showNextThumbnail: PropTypes.func,
    showPrevThumbnail: PropTypes.func
  }
  render() { 
    const { url, showNextThumbnail, showPrevThumbnail } = this.props
    const jumboStyle = {
      maxHeight: "450px"
    }
    return (
      <div className="jumbotron-container">
        <div className="arrow"  style={{float:"left"}}>
          <i onClick={showPrevThumbnail} className="fa fa-angle-left fa-5x" aria-hidden="true" />
        </div>
        <div className="jumbotron-main" style={{height:"500px"}}>
          <img src={url} style={jumboStyle} alt="no results" />
        </div>
        <div className="arrow" style={{float:"right"}}>
          <i onClick={showNextThumbnail} className="fa fa-angle-right fa-5x" aria-hidden="true" />
        </div>
      </div>
    );
  }
}

class ThumbnailRow extends Component {
  static propTypes = {
    thumbnails: PropTypes.array,
    displayThumbnail: PropTypes.func
  }
  render() {
    const { thumbnails, displayThumbnail } = this.props;
    return (
    <div className="thumbnail-row">
      {thumbnails.map((el, idx)=>{
        return (
          <Thumbnail
          key={idx} 
          idx={idx}
          url={el}
          displayThumbnail={displayThumbnail}
          />
        )
      })}
    </div>
    );
  }
}

class Thumbnail extends Component {
  static propTypes = {
    idx: PropTypes.number.isRequired,
    url: PropTypes.string,
    displayThumbnail: PropTypes.func
  }
  render() {
    const { idx, url, displayThumbnail } = this.props
    const thumbnailStyle = {
      boxSizing: "border-box",
      maxHeight: "150px",
      float:"left",
      padding: ".5em",
      textAlign: "center"
    }
    return (
      <div className="thumbnail">
          <img 
            src={url} 
            style={thumbnailStyle}
            alt={ "result #" + idx + 1 }
            onClick={() => { displayThumbnail(idx) }}
          />
      </div>
    );
  }
}

export const displayThumbnail = () => {};
export { App, Jumbotron, Thumbnail, ThumbnailRow }; 
