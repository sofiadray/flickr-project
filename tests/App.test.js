import React from 'react';
import ReactDOM from 'react-dom';
import { App, Jumbotron, Thumbnail, ThumbnailRow, displayThumbnail} from '../client/components/App';
import { shallow, mount, render, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';

var fetch = jest.fn(() => new Promise(resolve => resolve()));

const flickr = require('../client/flickr.js')
jest.mock('react-dom');
configure({ adapter: new Adapter() });

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
});

it('should render jumbo correctly', () => {
  const tree = renderer.create(
    <Jumbotron url="mockURL" />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

it('should render thumb correctly', () => {
  const tree = renderer.create(
    <Thumbnail onClick={displayThumbnail} idx={0} url="anotherMockURL"/>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

describe('getPhotos() using promises', () => {
  it('should not search without search term', () => {
   	var data = flickr.getPhotos('')
    return  expect(data).not.toBeDefined()
  })
})

it('renders list of thumbnails', () => {
  const element = (<Thumbnail url="mockURL" idx={0}/>);
  const wrapper = shallow(<ThumbnailRow thumbnails={["mockURL"]} />);
  expect(wrapper.contains(element)).toEqual(true);
});


