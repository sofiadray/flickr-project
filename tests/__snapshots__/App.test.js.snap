// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render jumbo correctly 1`] = `
<div
  className="jumbotron-container"
>
  <div
    className="arrow"
    style={
      Object {
        "float": "left",
      }
    }
  >
    <i
      aria-hidden="true"
      className="fa fa-angle-left fa-5x"
      onClick={undefined}
    />
  </div>
  <div
    className="jumbotron-main"
    style={
      Object {
        "height": "500px",
      }
    }
  >
    <img
      alt="no results"
      src="mockURL"
      style={
        Object {
          "maxHeight": "450px",
        }
      }
    />
  </div>
  <div
    className="arrow"
    style={
      Object {
        "float": "right",
      }
    }
  >
    <i
      aria-hidden="true"
      className="fa fa-angle-right fa-5x"
      onClick={undefined}
    />
  </div>
</div>
`;

exports[`should render thumb correctly 1`] = `
<div
  className="thumbnail"
>
  <img
    alt="result #01"
    onClick={[Function]}
    src="anotherMockURL"
    style={
      Object {
        "boxSizing": "border-box",
        "float": "left",
        "maxHeight": "150px",
        "padding": ".5em",
        "textAlign": "center",
      }
    }
  />
</div>
`;
